.github/workflows/validate.yml

```diff
@@ -64,19 +64,19 @@ jobs:
 
       - uses: actions/checkout@v2
 
-      - uses: haskell/actions/setup@v1
-        id: setup-haskell
-        with:
-          ghc-version: ${{ matrix.ghc }}
-          cabal-version: latest # default, keeping for visibility
+      - name: ghcup
+        run: |
+          ghcup install ghc ${{ matrix.ghc }}
+          ghcup set ghc ${{ matrix.ghc }}
+          cabal update
 
       #  See the following link for a breakdown of the following step
       #  https://github.com/haskell/actions/issues/7#issuecomment-745697160
       - uses: actions/cache@v2
         with:
           # validate.sh uses a special build dir
           path: |
-            ${{ steps.setup-haskell.outputs.cabal-store }}
+            ~/.cabal/store
```

Comment:
I am afraid this will not work in Windows
That alone is a good reason to use the action

Reply:
Yeah... however it looks to me like none of the caching is working anyway? All the runs I've checked say "cache not found".